### ISSUE TITLE
Project names must be all lowercase

### ISSUE THREAD

**Issue by glennerik** [rocjs/roc#92]
If you use any uppercase letters in your project name with roc init,
you don't get a warning that npm doesn't allow uppercase letters in the project name.
Instead you get an error at the end of the whole roc init process.
This can be confusing.

**Comment by dlmr** [comment]
I have now looked into this a bit more and I can't replicate this when using `roc init`. The current logic already handles this case and will automatically enforce lowercase for the project name.

This question will always produce a lowercase and dashed name that will be used by npm.

```
? What's the name of your application?
```

This is managed by the following filter functions

``` js
(input) => input.toLowerCase().split(' ').join('-')
```

https://github.com/rocjs/roc/blob/master/src/commands/helpers/default-prompt.js#L9

**Comment by glennerik** [comment]
Then it must have been coincidence that I experienced errors the same time as I used uppercase letters in the project name. I remember I tried this several times. I will be testing roc some more very soon, and I will get back with any information I find regarding this.

**Comment by tofagerl** [comment]
I get an error if using `'` at any point in the init/new process. That might have been it.

**Comment by dlmr** [comment]
This should be solved using the new template logic that has been merged in #115.

We now validate the name in the same way as npm does it.
